bin/generate.py

```diff
@@ -1043,10 +1043,13 @@ def generate_empty_interactive_sample_ans():
                 return True
             if not problem.interactive and not problem.multi_pass:
                 return True
-            for ext in ["statement", "download"]:
-                ans_ext_file = infile.with_suffix(f".ans.{ext}")
-                if infile.with_suffix(f".in.{ext}").is_file() and not ans_ext_file.is_file():
+            for ext in ["", ".statement", ".download"]:
```

Comment:
- the empty `.ans` files is not generated yet.
- the empty `.ans` file is explicitly not generated if `solution:` is set, do we want to skip this generation as well?

Reply:
@mpsijm ping

Reply:
- Good point, let's move the calling of this function to a later step :slightly_smiling_face: Do you have a multi-pass problem where `.ans` files are not empty, so I can test with that as well?
- Hmmm... that's two conflicting things :sweat_smile: Because if `solution:` is empty for the `sample` directory, but the problem is interactive and therefore requires an empty `.ans` file, I still think that this empty `.ans` file should be automatically generated?

Reply:
I don't have any problem where .ans is not "the default thingy" XD

but why would you set solution empty? (it is only used to generate the `.interaction` if there is none of `.statement` or `.out`)

Reply:
Yeah, the value of `solution` doesn't matter when both `.in` and `.interaction` already exist...

The samples are really weird compared to "normal" test cases, because samples work perfectly well without `.in` file. But things like `solution:` only make sense with a proper `.in` file, so if a sample test case doesn't have this, the value of `solution` is not used in any way. But we still want some automation for empty `.ans` files. :upside_down_face: 

I think I'm happy with how the code currently works, is there something you would like to see changed, or do you know a certain configuration that we should reject but currently don't, or a configuration that should work but currently doesn't?

Reply:
i would still want to skip empty `.ans` generation if the solution is explicitly set empty... However my only reason for this is backwards compatibility.